lute/require/src/bundlevfs.cpp

```diff
@@ -0,0 +1,127 @@
+#include "lute/bundlevfs.h"
+
+#include "Luau/Common.h"
+
+#include <string>
+
+static const std::string BUNDLE_PREFIX = "@bundle/";
+
+BundleVfs::BundleVfs(const Luau::DenseHashMap<std::string, std::string>& bundleMap)
+    : filePathToBytecode(bundleMap)
+{
+}
+
+static bool isBundleModule(const Luau::DenseHashMap<std::string, std::string>& bundleMap, const std::string& path)
+{
+    // Strip @bundle/ prefix if present
+    std::string lookupPath = path;
+    if (path.find(BUNDLE_PREFIX) == 0)
+        lookupPath = path.substr(BUNDLE_PREFIX.size());
+
+    // Check direct file match
+    if (bundleMap.find(lookupPath) != nullptr)
+        return true;
+
+    return false;
+}
+
+static bool isBundleDirectory(const Luau::DenseHashMap<std::string, std::string>& bundleMap, const std::string& path)
+{
+    // Strip @bundle/ prefix if present
+    std::string lookupPath = path;
+    if (path.find(BUNDLE_PREFIX) == 0)
```

Comment:
We don't have C++20's `starts_with` API, but we can improve this to a O(1) check for now. I think as future TODO, we could just have some string utilities exposed for this type of thing.
```suggestion
    if (path.rfind(BUNDLE_PREFIX, 0) == 0)
```
(same for other places in this file where we do `string.find(prefix) == 0`)